azure-ipam/ipam.go

```diff
@@ -132,8 +139,18 @@ func (p *IPAMPlugin) CmdDel(args *cniSkel.CmdArgs) error {
 	p.logger.Debug("Making request to CNS")
 	// cnsClient enforces it own timeout
 	if err := p.cnsClient.ReleaseIPAddress(context.TODO(), req); err != nil {
-		p.logger.Error("Failed to release IP address from CNS", zap.Error(err), zap.Any("request", req))
-		return cniTypes.NewError(cniTypes.ErrTryAgainLater, err.Error(), "failed to release IP address from CNS")
+		if errors.Is(err, &cnscli.ConnectionFailureErr{}) {
+			addErr := fsnotify.WatcherAddFile(args.ContainerID, watcherPath, watcherDirectory)
+			if addErr != nil {
+				p.logger.Error("Failed to add file to watcher", zap.Error(addErr))
+				return cniTypes.NewError(cniTypes.ErrTryAgainLater, err.Error(), "Failed to add file to watcher")
+			} else {
+				p.logger.Info("File successfully added to watcher directory")
```

Comment:
Should we log here that we failed to release IPs from CNS due to connection failure as well but is retriable once the connection is back? As currently in case of connection failure we will only see `success` log message( despite of the connection error.

Reply:
Add context as well in logs (like ipaddress/pod info both in info/success/failure logs)